### ISSUE TITLE
How to identify PII columns in a DataFrame.

### ISSUE THREAD

**Issue by kvrameshreddy** [microsoft/presidio#600]
While identifying and  anonymizing on text data presidio works wonderfully. Can we use the same methodology to identify PII columns in a Dataframe.

Thankyou.

**Comment by omri374** [comment]
Hi @kvrameshreddy,
We have a draft suggestion for this on this PR: https://github.com/microsoft/presidio/pull/436
The idea is to call the `analyze` method in batch. See this line for the specific method you could use:
https://github.com/microsoft/presidio/blob/c0b5dcf3320b281b78a01c9f012a15d47f5adf1c/presidio-analyzer/presidio_analyzer/analyzer_engine.py#L164

Would be happy to hear what you think of it and whether it solved your problem.

**Comment by kvrameshreddy** [comment]
Hi @omri374, 
"analyze_batch" will solve my problem. When can i expect the PR get merged in main branch.

And what if we have a "int" column in the DataFrame, How that column will be processed in this. Can you suggest any idea how to deal with this scenario.

Thankyou.

**Comment by omri374** [comment]
Hi @krameshreddy, this is still evolving and we're trying to better understand the requirements around this, so I don't have a final answer on when this will be part of the package.

The interface here lets you choose which columns you'd like to run this on. We'd be happy to hear more about your use case and how would you envision such a feature to look like. Adding @willsthompson as well who was actively working on this, to bring his perspective on the matter.

**Comment by kvrameshreddy** [comment]
Thanks for the reply @omri374.

My use case is to identify semantic data types(PII Data types Majorly ) of a Data frame based on their column values. 
The Data frame may consist of columns with strings like names, countries and all the PII data and columns with integers like age, weight etc., I am thinking like if we pass a Data frame as input the model should return a list of column types(recognized entities) and others as "str" and "int". 

Thankyou.

**Comment by willsthompson** [comment]
> And what if we have a "int" column

This is a good question. I think under @omri374's plan to use JSON for processing batch, it could potentially be handled as the [JSON Number type](https://tools.ietf.org/html/rfc7159#section-6). However, maybe for simplicity it would be better to assume everything is a string. I can see how having native numerical values might be helpful if you want to look for patterns that manifest statistically, but it seems like it could be quite disruptive to update the engine to support that all the way down (I think you would need to add something like a `NumericalAnalyzer` akin to `PatternAnalyzer`). 

The way I was thinking about this is to simply handle any statistical analysis upstream of Presidio and deal with strings for pattern matching. If it's in a DataFrame and Presidio is going to handle columns as JSON, it's going to be more efficient to use the DataFrame for stats. I think a compromise for Presidio might be a way to indicate "sidecar" data - for example, say you have a column `MyNum` and you pass it to Presidio either in full or a sample. It might be helpful to run some descriptive stats in Pandas and pass that into Presidio as well, so a pattern recognizer operating on `MyNum` could have access to those stats to validate/reject/boost/demote its score, but not treat it as an independent property/column.

**Comment by omri374** [comment]
We've created some documentation on this matter here: https://github.com/microsoft/presidio/discussions/714
Please share any feedback you have on this.

**Comment by AkshayDube11** [comment]
Hi Everyone, 

Note: I am writing on this thread as my queries are in line with it.

I have two queries in continuation with the above-mentioned discussion: 

1. How to use presidio with columns of dataframes having free text, also with a dataframe as a whole? My primary objective.
2. Can we set score_thresholds for different recognizers used? For Eg: How to have separate score_thresold values for entities such as PERSON, Location, etc. while pushing the free text through **the analyzer** step. 

To have a better reference to the questions above, my code snippet is as mentioned below:

step1: I converted my excel format dataframe column to a list with Dtype of string.
Code:
df1 = str(df.to_dict(orient="list"))

step2: I tried using a multilingual model as per my used case and I successfully got it working but I need a custom score_thresold setting as some of the Person Names are getting missed or wrong names are getting anonymized by the presidio anonymizer.
**Analyzer Code:**
analyzer_results = analyzer.analyze(text=df1, language='xx',entities=["EMAIL_ADDRESS", "PERSON", "LOCATION"], score_threshold=0.90)

print(analyzer_results)

**Anonymizer Code:**
anonymizer = AnonymizerEngine()

anonymized_results = anonymizer.anonymize(
    text=df1,
    analyzer_results=analyzer_results,    
    operators={"PERSON": OperatorConfig("replace", {"new_value": "<PER>"}),
               "EMAIL_ADDRESS": OperatorConfig("replace", {"new_value": "<EMAIL>"}),
               "LOCATION": OperatorConfig("replace", {"new_value": "<LOCATION>"})})
